File: DSAlgorithms/Leetcode/c++/sliding_window/658-find_k_closest_element.cpp
class Solution {
public:
    vector<int> findClosestElements(vector<int>& arr, int k, int x) {
       int left = 0;
       int right = arr.size() - k; 
       while(left < right)
       {
           int m = left + (right - left) / 2;
           if((x - arr[m]) > (arr[m + k] - x))
               left = m + 1;
            else
                 right = m;
       }
       std::vector<int> result;
       for(int i = left; i < left + k; i++)
            result.push_back(arr[i]);
       return result;
    }
};

File: DSAlgorithms/Leetcode/ts/src/sliding_window/658-find_k_closest_element.ts
const findClosestElements =(arr: number[], k: number, x: number): number[] =>{
    let left = 0;
    let right = arr.length - k;
    while(left < right)
    {
        let m = Math.floor((left + right) / 2);
        if((x - arr[m]) > (arr[m + k] - x))
            left = m + 1;
        else
            right = m;
    }
    const result: number[] = [];
    for(let i = left; i < left + k; i++)
        result.push(arr[i]);
    return result;

};